packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.tsx

```diff
@@ -165,14 +172,14 @@ export const SimpleFormIterator = (inProps: SimpleFormIteratorProps) => {
                             fields={fields}
                             getItemLabel={getItemLabel}
                             index={index}
-                            member={`${source}.${index}`}
+                            member={`${index}`}
                             onRemoveField={removeField}
                             onReorder={handleReorder}
                             record={(records && records[index]) || {}}
                             removeButton={removeButton}
                             reOrderButtons={reOrderButtons}
                             resource={resource}
-                            source={source}
+                            source={finalSource}
```

Comment:
We should stop passing the source altogether, and let the item read the sourceContext instead